src/app/shared/pipes/itemLink.ts

```diff
@@ -0,0 +1,12 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { ItemType, typeCategoryOfItem } from '../helpers/item-type';
+import { incompleteItemStringUrl } from '../routing/item-route';
+
+function transform({ id, type }: {id: string, type: ItemType}): any {
+  return incompleteItemStringUrl(id, typeCategoryOfItem({ type }));
+}
+
+@Pipe({ name: 'itemLink', pure: true })
+export class ItemLinkPipe implements PipeTransform {
+  transform = transform;
```

Comment:
why don't you just put the `transform` function in the class?

Reply:
I'm agree, i just did by analogy with user pipe, but there is helper function. I'll fix it

Reply:
fixed